ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -336,6 +337,16 @@ instance TPraosCrypto c => QueryLedger (ShelleyBlock c) where
       :: Query (ShelleyBlock c) result
       -> Query (ShelleyBlock c) (Serialised result)
 
+    GetFilteredRegisteredStakeCreds
+      :: Set (SL.Credential 'SL.Staking c)
+      -> Query (ShelleyBlock c) (SL.StakeCreds c)
+
+    GetFilteredDelegations
+      :: Set (SL.Credential 'SL.Staking c)
+      -> Query
+          (ShelleyBlock c)
+          (Map (SL.Credential 'SL.Staking c) (SL.KeyHash 'SL.StakePool c))
```

Comment:
@dcoutts: Do you think it's useful to filter based on a `Set` or should these just be simple lookups for a single stake credential?